### ISSUE TITLE
HeidiSQL Fails to Load dialog.dll Plugin

### ISSUE THREAD

**Issue by MattMencel** [HeidiSQL/HeidiSQL#54]
## Expected behavior

Connect to MariaDB with PAM authentication enabled.

## Current behavior

"Error: Authentication plugin 'dialog' cannot be loaded."

### Steps to reproduce

1. Downloaded the latest 64-bit version of HeidiSQL and installed on a Windows 2016 server.  Also tested on a Windows 10 desktop with the same result.
2. Setup a connection to the MariaDB server.
3. "Error: Authentication plugin 'dialog' cannot be loaded."

### Context

* HeidiSQL version: 9.5.0.5196
* Database system + version:  MariaDB 10.2.11 on CentOS 7
* Operating system:  Win2016 and Win10

**Comment by ansgarbecker** [comment]
`dialog.dll` is normally located in the `plugins` folder within the HeidiSQL folder. Can you verify it is there?
If it's there, can you have a look at the filesize, to be sure you have the current one: should be 55808 Bytes on a 64bit HeidiSQL install.

**Comment by mjmfighter** [comment]
I am getting this same error in HeidiSQL.  I am using the same version as he is (even did a clean install by removing all of HeidiSQL).  I am also running MariaDB 10.2.11 on CentOS 7.  

I have verified that the `dialog.dll` is in the `plugins` folder within the HeidiSQL folder.  In fact, the only two plugins in the folder are `dialog.dll` and `auth_gssapi_client.dll`

**Comment by ansgarbecker** [comment]
Probably the dialog.dll is outdated and should be taken from a newer MariaDB release. You can test yourself, by downloading the 64bit MariaDB, and copying the dialog.dll to HeidiSQL\plugins\.

**Comment by nerditup1** [comment]
We had this issue with HeidiSQL version 9.5.0.5196 installed on Windows 10.  I ran procmon and determined that heidisql.exe was looking for the dialog.dll file in the "C:\Program Files\MySQL\MySQL Server 5.6\lib\plugin" folder.  Obviously the dialog.dll file was not there because we are not running the mysql server on this Windows 10 machine.  I created a folder named "C:\Program Files\MySQL\MySQL Server 5.6\lib\plugin" and copied the dialog.dll file into that folder.  Now HeidiSQL is working as it should.

**Comment by ansgarbecker** [comment]
Ok, this lookup into "C:\Program Files\..." must be done by libmysql.dll. My code right before connecting to a server looks like this:
```pas
PluginDir := AnsiString(ExtractFilePath(ParamStr(0))+'plugins\');
mysql_options(FHandle, MYSQL_PLUGIN_DIR, PAnsiChar(PluginDir));
Connected := mysql_real_connect(...
```
So HeidiSQL _should_ just look in the plugins folder, nowhere else. When it does so, I suppose that's due to an incompatible dialog.dll.

**Comment by wcwood** [comment]
Yes, we had saw on the internet ...\MySQL Server 5.7\... and that didn’t work, so doing a trace on the executable we found that yes, it was looking off it in ...\MySQL Server 5.6\...

Thanks all!

W. Wood

On Mar 5, 2018, at 2:18 PM, nerditup1 <notifications@github.com<mailto:notifications@github.com>> wrote:


We had this issue with HeidiSQL version 9.5.0.5196 installed on Windows 10. I ran procmon and determined that heidisql.exe was looking for the dialog.dll file in the "C:\Program Files\MySQL\MySQL Server 5.6\lib\plugin" folder. Obviously the dialog.dll file was not there because we are not running the mysql server on this Windows 10 machine. I created a folder named "C:\Program Files\MySQL\MySQL Server 5.6\lib\plugin" and copied the dialog.dll file into that folder. Now HeidiSQL is working as it should.
